### ISSUE TITLE
Fedora 404 reported as a success

### ISSUE THREAD

**Issue by seth-shaw-unlv** [mjordan/riprap#48]
I was just checking my riprap database and noticed the following lines:
```
+------+--------------------------------------+------------+-----------------------------------------------------------------+--------------------------+------------------+------------------------------------------------------------------+-----------------------+---------------+---------------------------+
| id   | event_uuid                           | event_type | resource_id                                                     | timestamp                | digest_algorithm | digest_value                                                     | event_detail          | event_outcome | event_outcome_detail_note |
+------+--------------------------------------+------------+-----------------------------------------------------------------+--------------------------+------------------+------------------------------------------------------------------+-----------------------+---------------+---------------------------+
... trimmed ...
|  325 | b335d11e-99f9-4415-959e-1a14362b840a | fix        | http://localhost:8080/fcrepo/rest/masters/ent/ent000658-016.tif | 2019-04-08T13:49:20-0500 | sha256           | 6c1068258133b5cc140df346be0e162d7812e9abb71b2fd73e2133e73ab4b37c | Initial fixity check. | success       |                           |
|  326 | 4c3a9eb5-76b9-44f3-8ef3-031de40f8312 | fix        | http://localhost:8080/fcrepo/rest/masters/ent/ent000851-126.tif | 2019-04-08T13:49:22-0500 | sha256           | 404                                                              | Initial fixity check. | success       |                           |
|  327 | 0fd9bc98-95cb-4bee-aed0-3d4899a7a0c2 | fix        | http://localhost:8080/fcrepo/rest/masters/ent/ent000961-072.tif | 2019-04-08T13:49:22-0500 | sha256           | d6a10e5361b5ad75672a01e5fc93dbcba4202785a9b2f7b42c17284f96927458 | Initial fixity check. | success       |                           |
... trimmed ...
```
Hidden among successful records, number 326 has "404" as the sha256 digest, which is obviously wrong and suspiciously like an HTTP response code. Turns out that the corresponding Fedora URL "http://localhost:8080/fcrepo/rest/masters/ent/ent000851-126.tif" does, indeed, return a 404 response. 

So, Fedora returned a 404, which got recorded as the sha256 value and recorded as a successful "Initial fixity check." This is obviously incorrect.

**Comment by mjordan** [comment]
Yikes, thanks, I'll investigate.

**Comment by mjordan** [comment]
@seth-shaw-unlv simple mistake on my part, I returned the HTTP status code instead of `false` if the code returned by a request to Fedora wasn't `200`. When you get a minute, could you test the issue-48 branch? If Riprap can't find the resource in Fedora (i.e., it gets a `404`), Riprap will report a failed event in is execution summary and leave an entry in its log file indicating a 404.

**Comment by seth-shaw-unlv** [comment]
@mjordan I can probably do that, but I'm not sure when. Since I fixed all those broken Gemini records I don't get those messages anymore, so I'll have to manufacture a failure. I also need to test your branch for #50. It might be next week...

**Comment by mjordan** [comment]
No hurry, whenever you can if fine. Thanks! And thanks for finding these bugs!

**Comment by seth-shaw-unlv** [comment]
FINALLY got around to testing this. Works as expected. 👍 

A few other things I noticed:

1. We don't do any sanity checking for PluginFetchResourceListFromDrupal so that if the JSONAPI module hasn't been implemented (resulting in a 404) we simply crash on the first foreach. It would probably be good to include a check for any status other than 200 and report out with a message like "JSON:API doesn't appear to be working; please ensure it is enabled."
2. It only checks the first content type in the content type listing. So, even if I have multiple content types listed in my settings file only the first one gets checked.

**Comment by mjordan** [comment]
@seth-shaw-unlv thanks very much for reporting and testing. I'll merge the issue-48 branch now and open new issues for the two you identify above.

**Comment by mjordan** [comment]
Resolved with 62cd0f0d67f4f785f9afae46200108d7dd0bcdf4.
